ProcessOptimizer/model_systems/noise_models.py

```diff
@@ -1,33 +1,47 @@
 from abc import ABC, abstractmethod
 from typing import Callable, List, Union, Optional
 
-from scipy.stats import norm, uniform
+import numpy as np
 
 class NoiseModel(ABC):
     """
     Abstract class that is the basis for noise models.
     """
     def __init__(
-            self,
-            noise_size: Optional[float]):
+        self,
+        noise_size: Optional[float],
+        seed: Union[int, None] = 42,
```

Comment:
`Optional[X]` is a synonym for `Union[X,None]`, so you can use

```suggestion
from typing import Optional
...
        seed: Optional[int] = 42,
```
I don't know which is easier to read, which I think should be what decides which to use.

Reply:
It makes sense to me to stick with Optional as the type hint here.